conjur/argparsewrapper.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+
+"""
+Argparsewrapper module
+
+This module is a wrapper for the third-party argparse module used to parse
+commandline arguments. The module controls the types of errors and help screens
+we return according to the types of inputs received.
+"""
+
+import argparse
+import sys
+
+class ArgparseWrapper(argparse.ArgumentParser):
+    """
+    Wrapper to override default behavior of the argparse module.
+    """
+
+    def error(self, message):
+        """
+        This method handles the errors that originate from incorrect
+        commands.
+        """
+        sys.stderr.write('Error %s\n' % message)
+        self.print_help()
+        sys.exit(1)
+
+    @staticmethod
+    def _subcommand_error(message, help_screen):
+        """
+        This method handles the errors that originate from incorrect
+        subcommands/flags.
+        """
+        sys.stderr.write('Error %s\n' % message)
+        sys.stderr.write('%s\n' % help_screen)
+        sys.exit(1)
+
+    # pylint: disable=arguments-differ
+    def parse_args(self, args=None, namespace=None, resource=None):
+        args, arg_flag = self.parse_known_args(args, namespace)
+        resource = args.resource if args else None
+        if not arg_flag:
+            return args
+
+        msg = 'unrecognized arguments: %s'
+        err_msg = (msg % ' '.join(arg_flag))
+        if not resource:
+            self.error(err_msg)
+        # pylint: disable=protected-access
+        # Get the namespace of the particular resource
+        resource_namespace = self._subparsers._actions[0].choices.get(resource)
```

Comment:
In addition, this is a problematic line (we discussed about it). `actions[0]` is not safe (can cause `IndexError`). Usage of `.get` is great. I think we should externalize the line to a method